katdal/chunkstore_s3.py

```diff
@@ -166,29 +186,28 @@ def from_url(cls, url, timeout=10, extra_timeout=1, **kwargs):
         extra_timeout : int or float, optional
             Additional timeout, useful to terminate e.g. slow DNS lookups
             without masking read / connect errors (ignored if `timeout` is None)
+        token : str
+            Bearer token to authenticate
         kwargs : dict
-            Extra keyword arguments: config settings or create_client arguments
+            Extra keyword arguments (unused)
```

Comment:
Just my personal view on this: Either remove these, or log a warning if someone uses them. Providing an interface that doesn't do anything feels a bit off...

Reply:
It needs to be there at the moment, since it's part of the glorious free-for-all `katdal.open(**kwargs)` chain... The `kwargs` could contain keys such as `capture_block_id` and `stream_name` which are not meant for the `S3ChunkStore` but passed in nonetheless from the top.

There are better ways of doing this but the priority seems to be low.

Reply:
I've tried that before but Ludwig pointed out that removing it would break the way katdal.open works: it takes kwargs and passes them to multiple places that may have an interest in taking config from them. If one makes it a warning, then you end up getting the warning when kwargs are passed that are legitimate for other consumers.

I still don't like it because misspelt keyword args get silently dropped, but fixing that needs a higher-level redesign.